File: Dagger_Android/src/main/java/com/android/dagger_android/fragment/FragmentA.java
package com.android.dagger_android.fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.android.dagger_android.Student;
import com.android.dagger_android.databinding.FragmentABinding;

import javax.inject.Inject;

import dagger.android.support.AndroidSupportInjection;

public class FragmentA extends Fragment {
    private FragmentABinding mViewBinding;

    @Inject
    Student mStudent;

    @Override
    public void onAttach(@NonNull Context context) {
//        AndroidInjection.inject(this);
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        mViewBinding = FragmentABinding.inflate(inflater, container, false);
        return mViewBinding.getRoot();
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mViewBinding.tvInfo1.setText(mStudent.toString());
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mViewBinding = null;
    }
}

File: Hilt/src/main/java/com/android/hilt/ui/FragmentA.kt
package com.android.hilt.ui

import android.app.Application
import android.app.Dialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.android.hilt.IStorage
import com.android.hilt.Student
import com.android.hilt.Teacher
import com.android.hilt.databinding.FragmentABinding
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject
import javax.inject.Named

@AndroidEntryPoint
class FragmentA : Fragment() {
    private var mViewBinding: FragmentABinding? = null

    @Inject
    lateinit var mStudent1: Student

    @Inject
    lateinit var mStudent2: Student

    @Inject
    lateinit var mTeacher: Teacher

    @Named("SpStorage")
    @Inject
    lateinit var mStorage1: IStorage


    @Named("SpStorage")
    @Inject
    lateinit var mStorage2: IStorage

    @Named("DbStorage")
    @Inject
    lateinit var mStorage3: IStorage

    @Inject
    lateinit var mDialog: Dialog

    @Inject
    lateinit var mApplicationContext: Application

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mViewBinding = FragmentABinding.inflate(layoutInflater, container, false)
        initView()
        return mViewBinding?.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mViewBinding?.tvInfo1?.text = "mStudent1:$mStudent1"

        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmStudent2:$mStudent2")

        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmTeacher: $mTeacher")

        mStorage1.save()
        mStorage1.get("name")

        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmStorage1:$mStorage1")

        mStorage2.save()
        mStorage2.get("name")

        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmStorage2:$mStorage2")

        mStorage3.save()
        mStorage3.get("name")

        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmStorage3:$mStorage3")


        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmDialog:$mDialog")

        mViewBinding?.tvInfo1?.append("\n------------")
        mViewBinding?.tvInfo1?.append("\nmApplicationContext:$mApplicationContext")

        mViewBinding?.tvInfo1?.append("\napplication==mApplicationContext:"+(requireActivity().application==mApplicationContext))
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mViewBinding = null
    }

    private fun initView() {
        mViewBinding?.btn1?.setOnClickListener {
            mDialog.show()
        }
    }

}